### PR TITLE
[Azure Search] adding filter property for Autocomplete in SDK

### PR DESCRIPTION
**Pull request by ishansrivastava90**
- includes generated code from swagger spec from azure-rest-api-specs:master
- added custom code to AutocompleteParameters for POST flavor of the API
- added tests for the property

### CODE REVIEW

**Comment by ishansrivastava90** [comment]
@brjohnstmsft FYI.

File: src/SDKs/Search/DataPlane/Search.Tests/Tests/AutocompleteTests.cs
Comment (brjohnstmsft) [inline src/SDKs/Search/DataPlane/Search.Tests/Tests/AutocompleteTests.cs]: @ishansrivastava90 There is no top-level `baseRate` property anymore after my Complex Types changes (I'm assuming this is using the test `Hotel` type). Please pick a different property on which to filter. Sorry for the inconvenience.

File: src/SDKs/Search/DataPlane/Search.Tests/Tests/AutocompleteTests.cs
Comment (brjohnstmsft) [inline src/SDKs/Search/DataPlane/Search.Tests/Tests/AutocompleteTests.cs]: nit: extra line

File: src/SDKs/Search/DataPlane/Search.Tests/Tests/AutocompleteTests.cs
Comment (brjohnstmsft) [inline src/SDKs/Search/DataPlane/Search.Tests/Tests/AutocompleteTests.cs]: `baseRate` is going away

**Review by bitsforthought — CHANGES_REQUESTED** [review]

**Comment by brjohnstmsft** [comment]
@ishansrivastava90 The build failure appears to be a bug in another Azure service's SDK. I'll merge from master again to resolve it. Sorry for the inconvenience.

**Review by bitsforthought — APPROVED** [review]

**Comment by brjohnstmsft** [comment]
Will merge once builds are passing

**Comment by ishansrivastava90** [comment]
@brjohnstmsft @dsgouda The CI/Travis-CI check is failing consistently with the message, 
_"E: Failed to fetch https://packages.microsoft.com/ubuntu/14.04/prod/dists/trusty/main/binary-amd64/Packages.gz  Hash Sum mismatch
E: Some index files failed to download. They have been ignored, or old ones used instead.
The command "sudo apt-get update" failed and exited with 100 during." _

Any ideas as to get this to pass?

**Comment by dsgouda** [comment]
@ishansrivastava90 We're taking a look

**Comment by ishansrivastava90** [comment]
@dsgouda Both @brjohnstmsft  and I are going to be OOF for the 1st part of the next week. If the issue is determined and fixed before we are back, feel free to merge the PR.

**Comment by dsgouda** [comment]
@ishansrivastava90 Please pull lkatest changes from master and update this PR, that should fix the CI issues

**Pull request merged**
